src/RegistrationController.js

```diff
@@ -61,115 +61,77 @@ function (
-        }
-      });
-      this.model = new Model();
+        var form = new Form(this.toJSON());
 
-      var checkPasswordMeetComplexities = function (model, showError) {
-        var password = model.get('password') || '';
-        _.each(schema.passwordComplexity.enabledComplexities, function (complexityName) {
-          var ele = Okta.$('#password-complexity-' + complexityName);
-          var complexityValue = this.get(complexityName);
-          ele.removeClass('password-complexity-satisfied password-complexity-unsatisfied password-complexity-error');
-          if (PasswordComplexityUtil.complexities[complexityName].doesComplexityMeet(complexityValue, password, model)){
-            ele.addClass('password-complexity-satisfied');
-          } else if (showError) {
-            ele.addClass('password-complexity-error');
+        this.listenTo(form, 'saved', _.bind(function() {
+          var activationToken = this.get('activationToken');
+          if (activationToken) {
+            var authClient = this.appState.settings.authClient;
+            authClient.signIn({
+              token: activationToken
+            })
+            .then(_.bind(function(transaction) {
+              RouterUtil.routeAfterAuthStatusChange(this, null, transaction.data);
+            }, this));
           } else {
-            ele.addClass('password-complexity-unsatisfied');
+            this.appState.set('username', this.get('userName'));
+            this.appState.trigger('navigate', 'signin/register-complete');
           }
-        }, schema.passwordComplexity);
-      };
+        }, this.model));
 
-      var form = new Form(this.toJSON());
-      properties.each(function(schemaProperty) {
-        var inputOptions = RegistrationFormFactory.createInputOptions(schemaProperty);
-        var name = schemaProperty.get('name');
-        if (name === 'password' || name === 'login') {
-          inputOptions.events = {
-            'input': function () {
-              checkPasswordMeetComplexities(this.model);
-            }
-          };
-        }
-        if (name === 'password') {
-          inputOptions.events.focusout = function () {
-            checkPasswordMeetComplexities(this.model, true);
-          };
-        }
-        form.addInput(inputOptions);
-      });
-      form.add(PasswordComplexity.extend({passwordComplexity: schema.passwordComplexity}));
-      this.add(form);
-      this.footer = new this.Footer(this.toJSON());
-      this.add(this.footer);
+        properties.each(function(schemaProperty) {
+          var inputOptions = RegistrationFormFactory.createInputOptions(schemaProperty);
+          var subSchemas = schemaProperty.get('subSchemas');
+          var name = schemaProperty.get('name');
+          form.addInput(inputOptions);
+          if (name === 'password' && subSchemas) {
+            form.add(SubSchema.extend({id: 'subschemas-' + name, subSchemas: subSchemas}));
+          }
```

Comment:
Don't have much context, but why do we do this for ``password`` only?

Reply:
Only the password field has SubSchema property which holds the password complexity object.